gslib/utils/signurl_helper.py

```diff
@@ -84,6 +86,9 @@ def CreatePayload(client_id,
   if (billing_project is not None):
     signed_query_params['userProject'] = billing_project
 
+  if generation is not None:
+    signed_query_params['generation'] = generation
```

Comment:
So this is the core logic for us, and the API does the rest?

Reply:
Yep! That's exactly right.